### PR TITLE
fbinstant sdk support

### PR DESCRIPTION
**Pull request by nasimafeef**
script is moved to head and init is populated with function calls that are needed for game start

### CODE REVIEW

File: js/index.js [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
-exports.onReady = new lib.Callback();
+  this.init = function () {
+    var FBInstant = this.FBInstant;
 
-init();
+    FBInstant.initializeAsync()
+      .then(function () {
+      });
+    FBInstant.startGameAsync();
+    FBInstant.setLoadingProgress(100);
```
Comment (rampr) [inline js/index.js]: shouldn't these two functions move into the .then () ?

Comment (nasimafeef) [reply js/index.js]: yea. tried that. game was stuck on progress screen then.
working on it for proper flow.

**Review by rampr — APPROVED** [review]

File: js/index.js [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
+      return {
+        name: playerName,
+        id: playerId,
+        user_pic: playerPic,
+        cxt_type: contextType,
+        cxt_id: contextId
+      };
+  };
+
+  this.getUserPhoto = function () {
```
Comment (rampr) [inline js/index.js]: let's try to mimick FBInstant API as much as possible.

**Review by vivekb91 — APPROVED** [review]

File: js/index.js (hunk 2 above)
Comment (nasimafeef) [reply js/index.js]: yup

File: js/index.js [hunk 3]
```diff
@@ -1,16 +1,164 @@
...
+      if (cb) {
+        cb();
+      }
+    });
+  };
+
+  // Returns the entry point that the game was launched from.
+  this.getEntryPointAsync = function (cb) {
+    this.FBInstant.getEntryPointAsync().then(function (entry){
+      cb(entry);
```
Comment (rampr) [inline js/index.js]: why can't we return Promises itself ?

Comment (rampr) [reply js/index.js]: This will keep the application code cleaner

**Review by Javasamurai — APPROVED** [review]

File: js/index.js (hunk 3 above)
Comment (nasimafeef) [reply js/index.js]: yes. moving to that.

**Review by vivekb91 — APPROVED** [review]

**Comment by rampr** [comment]
looks good

**Review by rampr — APPROVED** [review]

**Comment by rampr** [comment]
Have a suggestion to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/prototype

Then this plugin will have very little code changes

**Pull request merged**